src/types.ts

```diff
@@ -242,9 +242,11 @@ export type UseProductSwatch = (props: UseProductSwatchProps) => ProductSwatchOb
 export interface ProductInfoObject {
   productSwatch: ProductSwatchObject | undefined;
   itemName: string;
+  itemId: string;
   itemPrice: number | undefined;
   itemUrl: string | undefined;
   itemImageUrl: string | undefined;
+  variationId: string | undefined;
```

Comment:
prefer `?` over explicitly defining `undefined`

```suggestion
  itemPrice?: number;
  itemUrl?: string;
  itemImageUrl?: string;
  variationId?: string;
```